nixos/modules/services/misc/devpi-server.nix

```
@@ -0,0 +1,125 @@
+{
+  pkgs,
+  lib,
+  config,
+  ...
+}:
+with lib;
+let
+  cfg = config.services.devpi-server;
+  serverDir = "/var/lib/devpi";
+in
+{
+  options.services.devpi-server = {
+    enable = mkEnableOption (mdDoc "Devpi Server");
+
+    package = mkOption {
+      type = with types; package;
+      default = pkgs.devpi-server;
+      defaultText = lib.literalExpression "pkgs.devpi-server";
+      description = "Which devpi-server package to use.";
+    };
+
+    primaryUrl = mkOption {
+      type = types.str;
+      description = "Url for the primary node. Required option for replica nodes.";
+    };
+
+    replica = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Run node as a replica.
+        Requires the secretFile option and the primaryUrl to be enabled.
+      '';
+    };
+
+    secretFile = mkOption {
+      type = types.str;
```

```suggestion
      type = types.nullOr types.path;
```

I think type here should be null or path.

---

I've refactored it to use null or path, but I'm unsure if this might mean that secrets will have to be stored in the nix store, which isn't ideal...